src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,224 @@
+        return input.startsWith("unmark");
+    }
+
+    /**
+     * Checks if the input is a "delete" command.
+     *
+     * @param input The user input.
+     * @return True if the input is a "delete" command, false otherwise.
+     */
+    public static boolean isDeleteCommand(String input) {
+        return input.startsWith("delete");
+    }
+
+    /**
+     * Parses the user command and performs the corresponding action.
+     *
+     * @param input The user input.
+     * @param tasks The list of tasks.
+     * @param ui    The user interface.
+     * @return True if the application should continue running, false if the user wants to exit.
+     * @throws DukeException If there is an error in processing the command.
+     */
+    public static boolean isValidCommand(String input, TaskList tasks, Ui ui) throws DukeException {
+
+        int taskIndex;
+        if (isDeleteCommand(input)) {
+            taskIndex = Integer.parseInt(input.substring(7).trim()) - 1;
+            if (taskIndex >= 0 && taskIndex < tasks.getSize()) {
+                tasks.removeTask(taskIndex);
+            }
+            return true;
+        } else if (isMarkCommand(input)) {
+            taskIndex = Integer.parseInt(input.substring(5).trim()) - 1;
+            if (taskIndex >= 0 && taskIndex < tasks.getSize()) {
+                tasks.markTask(taskIndex);
+            }
+            return true;
+        } else if (isUnmarkCommand(input)) {
+            taskIndex = Integer.parseInt(input.substring(7).trim()) - 1;
+            if (taskIndex >= 0 && taskIndex < tasks.getSize()) {
+                tasks.unmarkTask(taskIndex);
+            }
+            return true;
+        }
+
+        if (isByeCommand(input)) {
+            return false;
+        } else if (isListCommand(input)) {
+            ui.showTaskList(tasks);
+        } else if (isCreateTaskCommand(input)) {
+            Task newTask = parseCreateTaskInput(input);
+            tasks.addTask(newTask);
+        } else if (isFindCommand(input)) {
+            String keyword = input.substring(5).trim();
+            ArrayList<Task> matchingTasks = tasks.findTasksWithKeyword(keyword);
+            ui.showMatchingTasks(matchingTasks);
+        } else {
+            throw new DukeException("I'm sorry, but I don't know what that means :-(");
+        }
+        return true;
+    }
+
+    /**
+     * Converts a string representation of a task into a Task object.
+     *
+     * @param line The string representation of the task.
+     * @return The Task object.
+     * @throws DukeException If there is an error in parsing the task.
+     */
+    public static Task convertToTask(String line) throws DukeException {
+        String[] parts = line.split("\\|\\|");
+
+        if (parts.length < 5) {
+            throw new DukeException("Invalid task format in file: " + line);
+        }
+
+        String taskType = parts[0].trim();
+        String done = parts[1].trim();
```

Comment:
`done` is ambiguous. Perhaps `taskStatus` conveys the String contents more clearly?